client/src/placement-context.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import useSWR from "swr";
+import { PLACEMENT_ENABLED } from "./env";
+import { useUserData } from "./user-context";
+
+export interface Fallback {
+  copy: string;
+  click: string;
+  view: string;
+  image: string;
+  by: string;
+}
+
+export interface PlacementStatus {
+  copy: string;
+  click: string;
+  view: string;
+  fallback?: Fallback;
+  image: string;
+}
+
+export const PlacementContext = React.createContext<
+  PlacementStatus | undefined
+>(undefined);
+
+export function PlacementProvider(props: { children: React.ReactNode }) {
+  const user = useUserData();
+  const { data: pong } = useSWR<PlacementStatus>(
+    !PLACEMENT_ENABLED || user?.settings?.noAds ? null : "/pong/get",
+    async (url) => {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ keywords: [] }),
+      });
+
+      if (!response.ok) {
+        throw Error(response.statusText);
+      }
+
+      return (await response.json()) as PlacementStatus;
+    },
+    {
+      revalidateIfStale: false,
```

Comment:
This option means we won't reload the ad when doing client side nav (e.g. from the search box).

I don't think just changing this to `true` fixes things, though, because then (I think) we'll see a flash of the old ad while the new one is loading. Either we accept this, or the ad not changing, or we'll need to do something like (which I haven't tested):

```
const { data, isLoading } = useSWR<PlacementStatus>(
...
);
const pong = isLoading ? undefined : data
```